File: C++/Selection_sort.cpp
#include <iostream>


class Selection_sort{

int data [0x0F] = {566,66,9,6,3,1,23,36,4,88,96,100,200,1,7};

public:
    void selection(){
        int backward , _forward , minimum;
        std::cout<<"\nUnsorted data\n"<<std::endl;
        display();


        for(backward=0x00 ; backward <0x0F-0x01 ; backward++){
            minimum = backward;
            for(_forward = backward+0x01; _forward<0x0F ; _forward++){
                if(data[_forward]<data[minimum]){
                    minimum = _forward;
                    int temp = data[backward];
                    data[backward] = data[minimum];
                    data[minimum] = temp;
                }
            }
        }


        std::cout<<"\nSorted data\n"<<std::endl;
            display();


    }

public:
    void display(){
        for(int v = 0x00 ; v < 0x0F ; v++){
            std::cout<<data[v]<<" ";
        }

    }
};

int main(){
    Selection_sort selection;
    selection.selection();
return 0x00;
}

File: python/Selection_Sort.py
class Selection_sort:

    def selection(self):
        print("Selection sort algorithm")
        data = [556, 66, 9, 6, 3, 1, 23, 36, 4, 88, 96, 100, 200, 1, 7]
        print("Unsorted data\n"+str(data))

        for v in range(0x00, len(data) - 0x01, +0x01):
            minimum = v
            for x in range(v + 0x01, len(data), +0x01):
                if data[x] < data[minimum]:
                    minimum = x
                    temp = data[v]
                    data[v] = data[minimum]
                    data[minimum] = temp
        print("\n Sorted data \n"+str(data))


def main(self=None):
    self = ""
    selection = Selection_sort
    selection.selection(self)


if __name__ == "__main__":
    main("")
